pycbc/types/timeseries.py

```diff
@@ -273,8 +273,12 @@ def at_time(self, time, nearest_sample=False, interpolate=None):
 
         if nearest_sample:
             time += self.delta_t / 2.0
-
-        return self[int((time-self.start_time)*self.sample_rate)]
+        deltat = time - self.start_time
```

Comment:
I do agree with CodeClimate about the lines here. It's not clear why we need 3 return statements on successive lines, but it does become quite confusing what is going on here.

If this complexity *is* needed, can it be well commented (ie. why check for `__len__` on delta_t, that feels hacky! ... If you are asking, is deltat a numpy array there's clearer ways to do that) I'm guessing that you are wanting this to run on an array of times, rather than just a single value? If that's true the documentation does need updating, and it should be checked if the other code paths work in that way.

Reply:
So yes the reason to check for __len__ is to check if the list is a numpy array or something similar. The whole point of this change is to allow for checking multiple times with one function call. 
I am open to finding a way to reduce the complexity of this but not sure how to do so. Basically the current logic is:
Is the input time an array or a single value?
If Its an array, I had to add a check for an array of length 0, because there was an if it get an array of length 0 otherwise. 
If it is not an array, keep the original functionality.
I have been running PyCBC Live with this change in effect and it does not seem to have broken anything else.

Reply:
Does the time_slice method (https://github.com/gwastro/pycbc/blob/dc5ff851280c65879ecda96a4faee99d3431f1a0/pycbc/types/timeseries.py#L178), not do what you want? ... It's better to use (or adapt) existing methods, rather than trying to force behaviour into another, especially in the core PyCBC Classes.

Reply:
Perhaps not, if your list of times is a set of unconnected times.

Reply:
Also I would like to mention that checking if `__len__` exists is exctly what is done in the definitions of all the single detector stats in ranking.py

Reply:
Indeed, time_slice does not do what I need. The araay that gets passed in is an error of single detector trigger times which will not generally be a continuous stretch

Reply:
So I'm not opposed to having this work on a set of times, but we may need to consider whether this should still be called `at_time`; `at_times` would make more sense. It also then needs to work in *all* case if multiple values are given, and the code here will fail if given an array and `interpolate == 'linear'`. The doc strings must also accurately reflect what the code can do.

Reply:
(Or something along the lines of, we change this to `at_times` and make it work only on arrays. Then write an additional `at_time` method, which takes one value, converts to a length one array, and just calls on to `at_times` ... This should all also work with length 0 arrays, so I'm not sure yet why that needs to be a special case).

Reply:
Fundamentally I can not think of a way to write the at_time function that meets all of the following criteria:
Preserves original functionality when passed a single number (i.e. returns a single number and not a length 1 array)
Has the functionality I need of being able to check an array of times
Doesn't error when given an array of length 0 (this happens if no single detector triggers occur)
Has less than three return statements

I can make it so the three return statements aren't on three successive lines but I think the problem is larger than that.

Reply:
I like the idea of having seperate at_time and at_times methods. I can code that. 
The reason for the length 0 special case is because of the need to make indices into ints. The indices of the times to check are calculated using floats and then cast to to an int. The numpy ` astype(int) ` method gives an error if called by an array of length 0.

Reply:
I was getting an error if a length 0 array got passed in, which occurs if no single detector triggers happen in an analysis block. But I think that was actually happening with a previous version of this method, the check for 0 may not be necessary anymore.

Is it acceptable if I just leave the at_time method exactly as it was, and then add an at_times method that takes an array and has my desired functionality?

Reply:
I made a new commit doing what I suggested above. I returned at_time to its previous version, and made a seperate at_times method.

Reply:
Okay this looks good. I think the two could (should) be consolidated later, but I think it's now clear what to expect.